handler/handler.go

```diff
@@ -44,7 +76,128 @@ func (h *ReindexRequestedHandler) Handle(ctx context.Context, event *models.Rein
+				break
+			}
+
+			listLength := len(list.Items)
+
+			if listLength == 0 {
+				break
+			}
+
+			for i := 0; i < listLength; i++ {
+				datasetChan <- list.Items[i]
+			}
+
+			offset += DefaultPaginationLimit
+			totalDocs += listLength
+		}
+		log.Info(ctx, "total number of dataset docs retrieved", log.Data{"total_documents": totalDocs})
+	}()
+	return datasetChan
+}
+
+func retrieveDatasetEditions(ctx context.Context, datasetAPIClient clients.DatasetAPIClient, datasetChan chan dataset.Dataset, serviceAuthToken string) chan DatasetEditionMetadata {
+	editionMetadataChan := make(chan DatasetEditionMetadata)
+	go func() {
+		defer close(editionMetadataChan)
+		noOfConcurrentExtractions := getNoOfConcurrentExtractions(len(datasetChan), maxConcurrentExtractions)
+		var wg sync.WaitGroup
+		for i := 0; i < noOfConcurrentExtractions; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for dataset := range datasetChan {
+					if dataset.Current == nil {
+						continue
+					}
+					editions, err := datasetAPIClient.GetFullEditionsDetails(ctx, "", serviceAuthToken, dataset.CollectionID, dataset.Current.ID)
+					if err != nil {
+						logData := log.Data{
+							"dataset_id":    dataset.Current.ID,
+							"collection_id": dataset.CollectionID,
+						}
+						log.Error(ctx, "error retrieving editions for dataset", err, logData)
+					}
+					for i := 0; i < len(editions); i++ {
+						if editions[i].ID == "" || editions[i].Current.Links.LatestVersion.ID == "" {
+							continue
+						}
+						editionMetadataChan <- DatasetEditionMetadata{
+							id:        dataset.Current.ID,
+							editionID: editions[i].Current.Edition,
+							version:   editions[i].Current.Links.LatestVersion.ID,
+						}
+					}
+				}
+			}()
+		}
+		wg.Wait()
+	}()
+	return editionMetadataChan
+}
+
+func getAndSendDatasetURLsFromLatestMetadata(ctx context.Context, datasetAPIClient clients.DatasetAPIClient, editionMetadata chan DatasetEditionMetadata, serviceAuthToken string) chan string {
+	datasetURLChan := make(chan string)
+	go func() {
+		defer close(datasetURLChan)
+		var wg sync.WaitGroup
+		noOfConcurrentExtractions := getNoOfConcurrentExtractions(len(editionMetadata), maxConcurrentExtractions)
+		for i := 0; i < noOfConcurrentExtractions; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for edMetadata := range editionMetadata {
+					metadata, err := datasetAPIClient.GetVersionMetadata(ctx, "", serviceAuthToken, "", edMetadata.id, edMetadata.editionID, edMetadata.version)
+					if err != nil {
+						continue
+					}
+					// TODO: The dataset url should be sent to the content-updated topic here in the future
+					// But for the time being, we are going to extract the urls and print them
+					url := metadata.DatasetLinks.LatestVersion.URL
+					datasetURLChan <- url
+				}
+			}()
+		}
+		wg.Wait()
+	}()
+	return datasetURLChan
+}
+
+func logExtractedDatasetURLs(ctx context.Context, datasetURLChan chan string) {
```

Comment:
Do you need a TODO comment here about this function saying that it is temporary, and needs to be replaced with something that does something useful with the list of URL's and stops logging them ?

Reply:
Done